File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
// TC- O(n)
// SC- O(n)

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s(nums.begin(), nums.end());
        int longest_len = 0;

        for(int num:s){
            // process it as a sequence only if its prev number is not present in the set
            if(!s.count(num-1)){
                int curr_len = 1;
                while(s.count(num+curr_len))
                    curr_len++;
                if(curr_len>longest_len)
                    longest_len=curr_len;
            }
        }
        return longest_len;

    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
# TC- O(n)
# SC- O(n)

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        maxlen = 0

        # put nums into a set
        unique = set(nums)

        for num in unique:
            if num-1 not in unique: # seq starts here
                left = num
                right = num
                while(right+1 in unique):
                    right += 1
                maxlen = max(maxlen, right-left+1)
        
        return maxlen
